Pinta.Gui.Widgets/Widgets/ColorPickerDialog.cs

```diff
@@ -0,0 +1,998 @@
+		picker_surface_cursor.SetSizeRequest (pickerSurfaceDrawSize, pickerSurfaceDrawSize);
+		picker_surface_overlay.SetSizeRequest (pickerSurfaceDrawSize, pickerSurfaceDrawSize);
+		if (small_mode)
+			picker_surface_selector_box.SetOrientation (Orientation.Vertical);
+		else
+			picker_surface_selector_box.SetOrientation (Orientation.Horizontal);
+
+		hue_cps.SetSliderWidth (cps_width);
+		sat_cps.SetSliderWidth (cps_width);
+		val_cps.SetSliderWidth (cps_width);
+		r_cps.SetSliderWidth (cps_width);
+		g_cps.SetSliderWidth (cps_width);
+		b_cps.SetSliderWidth (cps_width);
+		a_cps.SetSliderWidth (cps_width);
+
+		sliders_box.Spacing = spacing;
+
+		this.GetContentAreaBox ().SetAllMargins (margins);
+
+		DefaultWidth = 1;
+		DefaultHeight = 1;
+	}
+
+	public void Setup ()
+	{
+		// Top part of the color picker.
+		// Includes palette, color surface, sliders/hex
+		// Basically, the not-swatches
+		top_box = new Gtk.Box { Spacing = spacing };
+
+
+		// titlebar of color picker; mainly just contains the reset color button
+		#region Titlebar
+
+		title_bar = new Gtk.HeaderBar ();
+		var reset_button = new Button ();
+		reset_button.Label = Translations.GetString ("Reset Color");
+		reset_button.OnClicked += (button, args) => {
+			colors = (Color[]) original_colors.Clone ();
+			UpdateView ();
+		};
+		title_bar.PackStart (reset_button);
+
+		var shrinkButton = new Gtk.Button ();
+		if (small_mode)
+			shrinkButton.SetIconName (Resources.StandardIcons.WindowMaximize);
+		else
+			shrinkButton.SetIconName (Resources.StandardIcons.WindowMinimize);
+		shrinkButton.OnClicked += (sender, args) => {
+			var contentArea = this.GetContentAreaBox ();
+			//contentArea.RemoveAll ();
+			SetSmallMode (!small_mode);
+			if (small_mode)
+				shrinkButton.SetIconName (Resources.StandardIcons.WindowMaximize);
+			else
+				shrinkButton.SetIconName (Resources.StandardIcons.WindowMinimize);
+		};
+
+		title_bar.PackStart (shrinkButton);
+
+		Gtk.Button ok_button = new Button { Label = Translations.GetString ("OK") };
+		ok_button.OnClicked += (sender, args) => { this.Response ((int) Gtk.ResponseType.Ok); this.Close (); };
+		ok_button.AddCssClass (AdwaitaStyles.SuggestedAction);
+
+		Gtk.Button cancel_button = new Button { Label = Translations.GetString ("Cancel") };
+		cancel_button.OnClicked += (sender, args) => { this.Response ((int) Gtk.ResponseType.Ok); this.Close (); };
+
+		title_bar.PackEnd (ok_button);
+		title_bar.PackEnd (cancel_button);
+
+		title_bar.SetShowTitleButtons (false);
+		this.SetTitlebar (title_bar);
+
+		#endregion
+
+
+		// Active palette contains the primary/secondary colors on the left of the color picker
+		#region Color Display
+
+		color_display_box = new Gtk.Box { Spacing = spacing };
+		color_display_box.SetOrientation (Orientation.Vertical);
+
+		var colorDisplayList = new Gtk.ListBox ();
+
+		if (colors.Length > 1) {
+			var colorDisplaySwap = new Gtk.Button ();
```

Comment:
Could you also add a tooltip for this button? I found it a bit hard to figure out from the icon what it did.
We could also add a better icon like this: https://fonts.google.com/icons?selected=Material+Symbols+Outlined:swap_vert:FILL@0;wght@400;GRAD@0;opsz@24&icon.query=swap&icon.size=24&icon.color=%23e8eaed which is from one of the icon sets we already use

Reply:
Added currently as `public const string EditSwap = "edit-swap-vert-symbolic";`. Thought the name fit best, but not sure.

Reply:
Thanks! That name sounds fine to me 👍